theano/tensor/basic.py

```diff
@@ -2588,6 +2588,175 @@ def nonzero_values(a):
     return a.flatten()[flatnonzero(a)]
 
 
+class UnravelIndex(gof.Op):
```

Comment:
basic.py is getting pretty long. I would put the new op in tensor/extra_ops.py to help split it. extra_ops is ops that isn't used very frequently.